catch-uncommitted.sh

```diff
@@ -33,13 +33,15 @@ done
 
 echo
 
+files="$(git ls-files --exclude-standard --others)"
 if [ $catch_no_git -eq 1 ] && !(hash git 2> /dev/null); then
 	echo "git not found, not running catch-uncommitted check"
 	exit 0
 elif ! git diff HEAD --exit-code ${ignore_eol} -- . "${extra_args[@]}" ; then
 	echo '** Uncommitted changes found (^^^ diff above ^^^) - FAIL **'
 	exit 1
-elif test -n "$(git ls-files --exclude-standard --others | tee /dev/tty)" ; then
+elif test -n "$files" ; then
+	echo $files
```

Comment:
Not sure if it makes a difference, but double quoting `"$files"` may or may not avoid a situation where one of the files is interpreted as a command-line option for the `echo` command (like a file whose name starts with the dash character). (Even with the the double quoting, this would not avoid the problem for the first file, but adding the double quotes is really cheap... I wouldn't go any further than adding the double quotes, given how theoretical / improbable this issue probably is.)

This is why I don't like shell scripts... :-) In other programming languages, there is no such ambiguity in even the most basic of constructs, like passing arguments to a "function".